src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -512,6 +512,23 @@ LoadShardIntervalList(Oid relationId)
 }
 
 
+/*
+ * LoadShardIntervalWithLargestShardId is a utility function that returns
+ * the largest shardId for the given relationId.
+ */
+ShardInterval *
+LoadShardIntervalWithLargestShardId(Oid relationId)
+{
+	CitusTableCacheEntry *cacheEntry = GetCitusTableCacheEntry(relationId);
+
+	int largestShardIndex = cacheEntry->shardIntervalArrayLength - 1;
+	ShardInterval *shardInterval =
+		CopyShardInterval(cacheEntry->sortedShardIntervalArray[largestShardIndex]);
```

Comment:
isn't this the shard with the largest hash range?

Reply:
I don't think we are interested in the hash ranges here? The idea is  to find the  shard name in  longest length, so  we  are searching some edge cases like the table has shards `table_99999` and `table_100000`, we want to  pick  the latter

Reply:
but this array is shorted by hash range

Reply:
but this array is sorted by (hash) range

which would no longer be the biggest shard ID after tenant isolation

Reply:
Oh, yes I see what you mean now

Reply:
We could be more defensive for future change and always loop over the shards, but that seems not very likely that we change the shardIds monotonically increasing per table?